File: web/src/types/contract.ts
export interface ContractDto {
  id: string;
  authorName: string;
  legalEntityName: string;
  description: string;
  createdDate: string;
  updatedDate?: string | null;
}

export interface CreateContractDto {
  authorName: string;
  legalEntityName: string;
  description: string;
}

export interface UpdateContractDto {
  id: string;
  authorName: string;
  legalEntityName: string;
  description: string;
}

File: bck/Core/Entities/Contract.cs
namespace Core.Entities;

public class Contract
{
    public Guid Id { get; set; }
    public required string AuthorName { get; set; }
    public required string LegalEntityName { get; set; }
    public required string Description { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime? UpdatedDate { get; set; }
}